### ISSUE TITLE
Improve implementation of seek

### ISSUE THREAD

**Issue by deepthidevaki** [zeebe-io/zeebe#6223]
Follow up #6216 

The implementation of seekToAsqn currently uses two seeks to get the correct behavior. This is not efficient. Find a better solution.

**Comment by npepinpe** [comment]
Being able to peek at the next entry would be an easy solution here. I think there might be other places where being able to peek at the next entry (i.e. being to tell if we will go over the commit index, for example) may be beneficial.

While peek is not part of the standard iterator pattern, it's pretty common in many places, so it wouldn't be entirely out of place - and with zero-copy serialization, I think it could be done in a cheap way (I guess not entirely free, but still).

**Comment by Zelldon** [comment]
is this something we should take a look at @deepthidevaki @npepinpe regarding performance hack week? Do you know the impact?

**Comment by npepinpe** [comment]
No, we don't have such fine grained performance metrics as far as I know. That said, I don't think seeking is part of the hot path anyway, since the common case would be to seek once then read sequentially.

**Comment by deepthidevaki** [comment]
`seek` is not in the hot path. It is used only once or twice when the streamprocessors starts after a role change. So it is not valuable to look at this in the hack week.
